### ISSUE TITLE
Mathjax version 3. Any update regarding font support other than MathJax TeX font?

### ISSUE THREAD

**Issue by gbala2412** [mathjax/MathJax#2503]
Mathjax version 3. Any update regarding font support other than MathJax TeX font?

**Comment by dpvc** [comment]
Work is progressing on this, but it is a complicated project.  Yesterday, I used a `stix2` branch that includes an SVG version of the STIX2 fonts.  There is a `stix2` subdirectory in that branch that includes an example HTML file that shows some stretchy characters, and an interactive file that you can use to type in your own TeX or MathML expressions for testing.  There is a `cli` directory that includes `tex2svg` and `mml2svg` commands that you cause to convert individual expressions to SVG files using the STIX2 fonts (and can use as templates for your own custom versions of these).  You can compare to the originals from the `direct` subdirectory of the [MathJax node demos repository](https://github.com/mathjax/MathJax-demos-node) to see what modifications are needed.

Finally, there is a `tables` directory that includes font tables for the STIX2 fonts as I've mapped them to the various `mathvariant` values for MathJax (variants starting with `-` are internal use to access things like the larger sizes of stretchy delimiters, calligraphic characters, and so on).

This branch currently is intended for server-side pre-processing only, as the data for the STIX2 fonts is very large, and needs to be broken in to pieces that can be loaded dynamically as needed in order to be used effectively in a browser.  Then CommonHTML support will follow relatively quickly.

The tools for producing the files needed by MathJax are coming along well, but are not yet ready for public use.

So while much work has been happening behind the scenes, it is not quite ready yet for prime time.  Work is going to slow down as I begin teaching again this fall, so a formal release will probably need to wait until early next year.

**Comment by gbala2412** [comment]
Thank you very much for your reply.

**Comment by BrackXs** [comment]
Are there any news? (Since it is 'early next year' now...)

**Comment by dpvc** [comment]
We are actively working on it.  There is a stix2 branch that includes an SVG version of the STIX2 fonts for server-side use (not yet for in-browser use) as an illustration of the effort.  We expect to include more fonts in a public release during the summer.

**Comment by pkra** [comment]
@BrackXs just a thought: if @le-tex actually supported mathjax, development might just be faster.

**Comment by kno10** [comment]
High on my font wish list are serif fonts, such as newtxsf which plays nicely with the popular Fira Sans font.

**Comment by imjasonmiller** [comment]
@kno10 I think you mean sans-serif? I wondered if these changes would make it possible to make use of [Fira Math](https://github.com/firamath/firamath) in the future?

**Comment by kno10** [comment]
Yes, sans serif obviously. Fira Sans is pretty popular, and I'd like math typography consistent with Fira Sans.
Fira Math seems to be work in progress. newtxsf supposedly is quite complete and similar enough for most use cases.

**Comment by apolanco115** [comment]
Can we expect Gyre Pagella font in v3?

**Comment by dpvc** [comment]
@apolanco115, yes.  I have the data files ready for Gyre Pagella read to go for the next release (and will have the other Gyre fonts as well).  The tools for creating the needed MathJax data files will be made available, so you could create your own font  for Fira Sans, or a font extension to override some of an existing font.

**Comment by giancarloantonucci** [comment]
Is there an approximate timeplan for this release? Sorry for pressuring, but loading custom fonts would truly be a blessing.

**Comment by dpvc** [comment]
I expect the font release to be in early Q2 this year.

**Comment by stnor** [comment]
@dpvc Any update on the font support?

**Comment by stnor** [comment]
Thanks! I'm still running beta3 in prod, and got some help back then from you with sans-fonts, so looking forward to upgrade when done! :)

**Comment by baptiste** [comment]
@dpvc I'm making _slow_ progress on the [single-stroke math font](https://www.bapt.xyz/b/handwriting) idea, but it would be brilliant if the new tools & infrastructure allowed this possibility (for experimentation) as well. 
I bet others would follow and draw their own glyphs, once the opportunity to typeset them via Mathjax is demonstrated.

![GRQM](https://user-images.githubusercontent.com/18970/168403550-efa6dc43-aa6a-4621-b7ab-f7dbdfbc23f3.gif)

I imagine having the full list of glyphs as a json dictionary containing the svg paths and their placement offsets would be a suitable input format?

**Comment by kno10** [comment]
The 4.0.0 Alpha.1 milestone had in the changelog:

* Support for 11 different fonts, including the STIX2 font set, five Gyre math fonts, and a sans-serif font based on the Fira-Math font.

So I guess there has been some progress?

**Comment by stnor** [comment]
Yup, I am using the Fira font in production on alpha.1 since a while back (simple use case - see https://nomp.se/nomp/matematik/93/algebra/forenkling-av-polynom/93.ALGEBRA.9)

**Comment by dpvc** [comment]
@kno10, yes, the v4 alpha release has the 11 fonts, and has been well received over all.  There are still some issues to resolve, however, and code review is not yet complete (but well underway).  I'm hoping a beta release will be soon, followed by the official release shortly thereafter.

**Comment by kno10** [comment]
close this, as additional fonts are available in 4.0? Works well for me, specific issues are likely best tracked in their own reports?

**Comment by dpvc** [comment]
The official version 4.0 has been released with font support.
